### PR TITLE
Reactor front end

### PR DESCRIPTION
**Pull request by bbaayezi**
- Refactored all components under `comments` folder with function component
- Add global state management support using react hooks `context` API
- Add a new global axios instance called `api`, back end address will be decided by the `NODE_ENV`

---
State management is still a problem, in this version of implementation the comments page will refresh after each `POST` request, this is the easiest way to sync the states on the front end with the data on the back end (unless we found a more suitable way to manage states in the future).

### CODE REVIEW

**Comment by bbaayezi** [comment]
Just came up with an idea, maybe we could use `websocket` for the comments page so that we can get real-time responses from the server, and no need to refresh the page all the time

**Review by dinojs — APPROVED** [review]

**Review by lukasotocerny — CHANGES_REQUESTED** [review]
Looks good. Just few structural change I'd suggest.
1. Renaming `CommentBox.jsx` to `CommentTab.jsx` and move it to `project-tabs` directory. So that it's semantically on the same level as `QRTab` and `DownloadTab` components.
2. I'd rename all uses of `Comment` to `Feedback`, e.g. `CommentForm` to `FeedbackForm`, or `CommentTab` to `FeedbackTab`. Since we're using word *Feedback* everywhere, I'd like to stick to that and it'd be only matter of time we'd have to do this change, or we'd completely diverge with the terminology.

File: src/components/comments/CommentBox.jsx [hunk 1]
```diff
@@ -1,215 +1,168 @@
...
+  useEffect(() => {
     // check if there is a user credential stored in localStorage
     // if not, create a new user credential
-    let storedUserStr = localStorage.getItem('user');
+    const storedUserStr = localStorage.getItem('user');
+    let userID = "";
+    let userName = "";
     if (storedUserStr === null) {
-      this.createAnonymousUser();
+      createAnonymousUser();
```
Comment (lukasotocerny) [inline src/components/comments/CommentBox.jsx:20]: Also, could we move this logic of creating user to `App.js` or somewhere global? I need `userId` even for building of projects.

**Comment by bbaayezi** [comment]
`FeedbackContextProvider` now wraps up the whole project. Also, since other pages also want to access the global context, and the default router will refresh the page and thus lost all global context, the default router will be changed to hash router.

**Review by lukasotocerny — APPROVED** [review]
really nice, LGTM! good that you have an eye on Dino's PR, so that yours don't clash with each other

**Pull request merged**